examples/ex10p.cpp

```diff
@@ -627,10 +627,7 @@ double HyperelasticOperator::ElasticEnergy(const ParGridFunction &x) const
 
 double HyperelasticOperator::KineticEnergy(const ParGridFunction &v) const
 {
-   double loc_energy = 0.5*M.InnerProduct(v, v);
-   double energy;
-   MPI_Allreduce(&loc_energy, &energy, 1, MPI_DOUBLE, MPI_SUM,
-                 fespace.GetComm());
+   double energy = 0.5*M.InnerProduct(v, v);
```

Comment:
I am also concerned that there is a hidden `ParallelProject` in the new version, so the new version is not really the same here (?)

Reply:
Solved by distinguishing between `InnerProduct()` and `TrueInnerProduct()`. We just collided in the development/review process :smile: .